lizmap/modules/lizmap/lib/Project/Project.php

```diff
@@ -467,6 +467,36 @@ public function getData($key)
         return $this->qgis->getData($key);
     }
 
+    /**
+     * Get the minimum needed project information for some pages (landing page, admin project listing).
+     *
+     * @return array of project metadata
+     */
+    public function getMetadata()
```

Comment:
I really want an object, not an array as a returned value. With an array, it's hard elsewhere in the code, to identify what it contains, where it is build etc.. I doesn't help to maintain the code. With a class/object, and with the help of an editor, we don't have such issues. We can have typed parameters on methods we gives this object etc.. So we have more robust code. 

I really want to eradicate all these kind of arrays we have in our code.

Reply:
Ok, I will add a new class projectMetadata

Reply:
The new class should have getter methods. To ease the usage, the constructor may accept an array like you build..
Something like this:
```
namespace Lizmap\Project;
class ProjectMetadata {

  function __construct($repository, $id, $metadata);

  function getTitle() {... }
...
}
```